runtime/util/strhelp.c

```diff
@@ -34,3 +34,18 @@ strnrchrHelper(const char *str, char c, UDATA len)
 	}
 	return ret;
 }
+
+BOOLEAN
+isStrSuffixHelper(const char* str, UDATA strLen, const char* suffix, UDATA suffixLen)
+{
+	BOOLEAN ret = FALSE;
+
+	if ((NULL != str) && (NULL != suffix) && (suffixLen <= strLen)) {
+		char* suffixStart = (char*)str + strLen - suffixLen - 1;
```

Comment:
Is this correct ? If `strlen == suffixLen`, the comparision starts from `suffixStart` which is `(char*)str - 1`.

Reply:
Right good catch. The condition here was supposed to be `(suffixLen < strLen)`.

Reply:
In our use case, the incoming `strLen` is inclusive of a null-terminator. If the `-1` isn't there, the `memcmp` is off by one. I also don't like the `-1`. An alternative approach would be to specify the API to only work with null-terminated strings, then call `strlen()` in the function for both, then we could avoid that at the extra cost of two `strlen()` on each call. Alternatively, in the current use case, I could `-1` from the caller side and fix up the function.

Reply:
Either of those are better. I did it the current way initially because that is also what the `strnchrHelper` function does.

Reply:
Please add a doxygen comment on function so that people know whether the strLen/suffixLen is supposed to include the null-terminator. I suggest not including the null-terminator for both of them as UTF-8 might not be null terminated.

Reply:
Will do. I have a doxygen comment over the declaration in `util_api.h`. Is it sufficient to just update that one or should I duplicate the comment over the definition too?

Reply:
> Is it sufficient to just update that one.

Updating that one should be sufficient.

Reply:
Done, thanks for pointing that out.